File: project_euler_rb/problem_036.rb
# The decimal number, 585 = 10010010012 (binary), is palindromic in both bases.

# Find the sum of all numbers, less than one million, which are palindromic in base 10 and base 2.

# (Please note that the palindromic number, in either base, may not include leading zeros.)

a = 1.upto(1_000_000).select do |num|
  next if num.to_s(2).start_with?('0')
  num.to_s == num.to_s.reverse && num.to_s(2) == num.to_s(2).reverse
end

puts "Answer: #{a.sum}"

# Complete!
# Answer = 872187

File: project_euler_py/problem_036.py
"""The number 585, 1001001001 in binary, is palindromic in both bases.

Find the sum of all numbers, less than one million,
which are palindromic in base 10 and base 2.

(Please note that the palindromic number, in either base,
may not include leading zeros.)
"""

import commons

answer = 0
for i in range(1, 1000000):
    b_i = int(bin(i)[2:])
    if commons.is_palindrome(i) and commons.is_palindrome(b_i):
        answer += i

print(answer)

# Answer: 872187
# Time: 0m1.830s
